From c740a40f3cb63d7416d60d28fe438791d3d3d0c7 Mon Sep 17 00:00:00 2001
From: Erik Noren <enoren@gmail.com>
Date: Wed, 15 Nov 2017 18:44:31 -0600
Subject: [PATCH] ISSUE #298 - Pass through the default value for fields that
 are not of union type so the default value is retained

---
 .../hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java b/schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java
index ed5c5ad..248290a 100644
--- a/schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java
+++ b/schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java
@@ -171,7 +171,7 @@ public class AvroSchemaResolver implements SchemaResolver {
                 updatedFields.add(new Schema.Field(field.name(),
                                                    fieldSchema,
                                                    field.doc(),
-                                                   currentFieldTypeIsUnion ? NullNode.getInstance() : null,
+                                                   currentFieldTypeIsUnion ? NullNode.getInstance() : field.defaultValue(),
                                                    field.order()));
             }
 
-- 
1.8.4.2

